xwiki-pro-macros-api/src/main/java/com/xwiki/macros/showhideif/internal/macro/AbstractShowHideIfMacro.java

```diff
@@ -125,6 +126,30 @@ protected boolean doesMatch(ShowHideIfMacroParameters parameters)
             default:
                 break;
         }
+
+        if (parameters.getDisplayType() != null
+            && parameters.getDisplayType() != ShowHideIfMacroParameters.DisplayType.NONE)
+        {
+            boolean isExportPrintable = "export".equalsIgnoreCase(xcontext.getAction());
+            switch (parameters.getDisplayType()) {
+                case DEFAULT:
+                    if (isExportPrintable) {
+                        matchAllRes = false;
+                    } else {
+                        matchAnyRes = true;
```

Comment:
Shouldnt it be something like this?

```suggestion
                        matchAllRes = false;
                        matchAnyRes = false;
                    } else {
                        matchAllRes = true;
                        matchAnyRes = true;
```